File: local_examples/client-specific/redis-py/landing.py
# EXAMPLE: landing
# REMOVE_START
import redis
# REMOVE_END

# STEP_START connect
r = redis.Redis(host='localhost', port=6379, decode_responses=True)
# STEP_END

# STEP_START set_get_string
r.set('foo', 'bar')
# True
r.get('foo')
# bar
# STEP_END

# STEP_START hash_operations
r.hset('user-session:123', mapping={
    'name': 'John',
    "surname": 'Smith',
    "company": 'Redis',
    "age": 29
})
# True

r.hgetall('user-session:123')
# {'surname': 'Smith', 'name': 'John', 'company': 'Redis', 'age': '29'}
# STEP_END

# STEP_START close
r.close()
# STEP_END

File: local_examples/client-specific/ioredis/landing.js
// EXAMPLE: landing
// STEP_START connect
import { Redis } from 'ioredis';

const redis = new Redis();
// STEP_END

// STEP_START set_get_string
await redis.set('key', 'value');
const value = await redis.get('key');
console.log(value); // >>> value
// STEP_END

// STEP_START set_get_hash
await redis.hset('user-session:123', {
    name: 'John',
    surname: 'Smith',
    company: 'Redis',
    age: 29
});

const userSession = await redis.hgetall('user-session:123');
console.log(JSON.stringify(userSession, null, 2));
/* >>>
{
  "surname": "Smith",
  "name": "John",
  "company": "Redis",
  "age": "29"
}
 */
// STEP_END

// STEP_START close
redis.disconnect();
// STEP_END
